infra/core/search/search-services.bicep

```diff
@@ -62,3 +62,4 @@ resource search 'Microsoft.Search/searchServices@2021-04-01-preview' = {
 output id string = search.id
 output endpoint string = 'https://${name}.search.windows.net/'
 output name string = search.name
+output principalId string = search.identity.principalId
```

Comment:
We try to keep our changes synced to azure-dev repo, so I've sent this as a PR https://github.com/Azure/azure-dev/pull/3325